### ISSUE TITLE
Cannot read property 'defineProperty' of undefined   webpack

### ISSUE THREAD

**Issue by penguinab** [babel/babel#12087]
## Bug Report
project with webpack、@babel/preset-env、@babel/plugin-transfrom-runtime、@babel/runtime-corejs3
project after bundled by webpack run in browser whith flowing Error (user preset-env plugin-transfrom-runtime)
```js
Uncaught TypeError: Cannot read property 'defineProperty' of undefined
    at Object../node_modules/core-js-pure/es/object/define-property.js (define-property.js:11)
    at __webpack_require__ (bootstrap:789)
    at fn (bootstrap:100)
    at Object../node_modules/core-js-pure/features/object/define-property.js (define-property.js:1)
    at __webpack_require__ (bootstrap:789)
    at fn (bootstrap:100)
    at Object../node_modules/@babel/runtime-corejs3/core-js/object/define-property.js (define-property.js:1)
    at __webpack_require__ (bootstrap:789)
    at fn (bootstrap:100)
    at Object../node_modules/@babel/runtime-corejs3/helpers/createClass.js (createClass.js:1)
```
```
require('../../modules/es.object.define-property');

var path = require('../../internals/path');

var Object = path.Object;
//debugger in this;  path.Objcet is undefined

var defineProperty = module.exports = function defineProperty(it, key, desc) {
  return Object.defineProperty(it, key, desc);
};

if (Object.defineProperty.sham) defineProperty.sham = true;
```

**Current behavior**
- [GitHub](https://github.com/penguinab/pen_webpack)

it build completes, but it emits  error in browser
it's ok if  just use @babel/preset-env with setting ("useBuiltIns":"usage" and "corejs":3)  without @babel/plugin-transfrom-runtime

**Expected behavior**
expect can extract  babel utils function  by using @babel/plugin-transfrom-runtime and @babel/runtime-corejs3  

**Babel Configuration (babel.config.js, .babelrc, package.json#babel, cli command, .eslintrc)**
- Filename: `babel.config.js`

```js
{
    "presets": [
        ["@babel/preset-env",{
            "debug": true
        }]
    ],
    "plugins": [
        [
            "@babel/plugin-transform-runtime",
            {
                "helpers": true,
                "regenerator": true,
                "corejs": 3,
                "absoluteRuntime": true
            }
        ]
    ]
}
```

- Filename: `webpck.config.js`
```js
{
        entry:{
            'index':path.resolve(__dirname,'../src/main.js'),
        },
        output:{
            path:path.resolve(__dirname,'../dist'),
        },
        module:{
            rules:[
                {
                    test:/\.js$/i,
                    exclude:'/node_modules/',
                    use:{
                        loader:'babel-loader',
                    }
                },
                {
                    test:/\.(png|jpg|gif)$/i,
                    use:{
                        loader:'url-loader',
                        options:{
                            limit: 10 * 1024,  //10kb
                            outputPath:'images/',
                            name:'[name]_[hash:8].[ext]'
                            // fallback: require.resolve('responsive-loader'), //压缩图片
                            // quality: 70
                        }
                    }
                },
                {
                    test:/\.[svg|ttf|woff|eot]/i,
                    use:[{
                        loader:'file-loader',
                        options:{
                            outputPath:'font/'
                        }
                    }]
                }
            ]
        },
        plugins:[
            new ProgressPlugin(),
            new HtmlWebpackPlugin({
                title:'penguin web',
                template:path.resolve(__dirname,'../template.html'),
                meta: {
                    viewport: 'width=device-width'
                },
                filename:'index.html',
        ]
    }

```
- Filename: `package.json`
```js
"devDependencies": {
    "@babel/cli": "^7.11.6",
    "@babel/core": "^7.11.6",
    "@babel/plugin-transform-runtime": "^7.11.5",
    "@babel/preset-env": "^7.11.5",
    "autoprefixer": "^10.0.0",
    "babel-loader": "^8.1.0",
    "clean-webpack-plugin": "^3.0.0",
    "copy-webpack-plugin": "^6.1.0",
    "css-loader": "^4.3.0",
    "file-loader": "^6.1.0",
    "html-loader": "^1.3.0",
    "html-webpack-plugin": "^4.4.1",
    "less": "^3.12.2",
    "less-loader": "^7.0.1",
    "mini-css-extract-plugin": "^0.11.2",
    "optimize-css-assets-webpack-plugin": "^5.0.4",
    "postcss": "^8.0.5",
    "postcss-loader": "^4.0.2",
    "responsive-loader": "^2.1.1",
    "style-loader": "^1.2.1",
    "terser-webpack-plugin": "^4.2.1",
    "url-loader": "^4.1.0",
    "webpack": "^4.44.1",
    "webpack-cli": "^3.3.12",
    "webpack-dev-server": "^3.11.0",
    "webpack-merge": "^5.1.4"
  },
"dependencies": {
    "@babel/runtime-corejs3": "^7.11.2"
  }
```

**Environment**

```
System:
    OS: Windows 7 6.1.7601
  Binaries:
    Node: 10.15.3 - C:\dev\nodejs\node.EXE
    Yarn: 1.17.3 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 6.4.1 - C:\dev\nodejs\npm.CMD
  npmPackages:
    @babel/cli: ^7.11.6 => 7.11.6
    @babel/core: ^7.11.6 => 7.11.6
    @babel/plugin-transform-runtime: ^7.11.5 => 7.11.5
    @babel/preset-env: ^7.11.5 => 7.11.5
    @babel/runtime-corejs3: ^7.11.2 => 7.11.2
    babel-loader: ^8.1.0 => 8.1.0
    webpack: ^4.44.1 => 4.44.1
```
- How you are using Babel: [e.g. `cli`, `webpack-loader']

**Possible Solution**
？？？

**Additional context**
Add any other context about the problem here. Or a screenshot if applicable

**Comment by penguinab** [comment]
i fixed this by reconstruct my project,it maybe  problem with npm pkg download

**Comment by JLHwung** [comment]
@penguinab Thanks for the reproduction repo. I can reproduce on my local machine but I could not figure out what is go wrong in `core-js`: It seems the https://github.com/zloirock/core-js/blob/master/packages/core-js-pure/override/internals/export.js is not working properly and thus `require("../internal/path")` does not return an `{ Object: Object }`. You can open an issue on the `core-js` project.

Not related: `exclude:'/node_modules/',` in your webpack config should be `exclude: /node_modules/`, otherwise the `@babel/runtime` will be transpiled by Babel.

**Comment by JMarkoski** [comment]
This was a difficult one to track down. Here is what happened, and how to solve it:

> **TL;DR**:
> 
> Your webpack config test for `file-loader` is wrong. Change `test:/\.[svg|ttf|woff|eot]/i` to `test: /\.(svg|ttf|woff|eot)$/i` and everything will work as expected.


First, `@babel/plugin-transform-runtime` as you configured it with the `corejs: 3` option and `@babel/runtime-corejs3`, adds polyfills that are not globally polluting. In this case it is for `Object.defineProperty`. More specifically, the polyfill that gets added is the following one: 

`"./node_modules/@babel/runtime-corejs3/core-js/object/define-property.js"`

which requires:

`"./node_modules/core-js-pure/features/object/define-property.js"`

which requires:

`"./node_modules/core-js-pure/es/object/define-property.js"`

which in the compiled webpack bundle has this line:

`__webpack_require__(/*! ../../modules/es.object.define-property */ "./node_modules/core-js-pure/modules/es.object.define-property.js");`

Now, first, why does it have this line? Because the code for `"./node_modules/core-js-pure/es/object/define-property.js"` looks like this:

```
require('../../modules/es.object.define-property'); // <= NOTICE THIS LINE HERE
var path = require('../../internals/path');

var Object = path.Object;

var defineProperty = module.exports = function defineProperty(it, key, desc) {
  return Object.defineProperty(it, key, desc);
};

if (Object.defineProperty.sham) defineProperty.sham = true;
```

The code that is required in that line is the following one:

```
var $ = require('../internals/export'); // <= NOW NOTICE THIS LINE
var DESCRIPTORS = require('../internals/descriptors');
var objectDefinePropertyModile = require('../internals/object-define-property');

// `Object.defineProperty` method
// https://tc39.github.io/ecma262/#sec-object.defineproperty
$({ target: 'Object', stat: true, forced: !DESCRIPTORS, sham: !DESCRIPTORS }, {
  defineProperty: objectDefinePropertyModile.f
});
```
Now the code in `'../internals/export'` has code that modifies the `path` object that you use in the these lines in one of the snippets above, for it ti be able to not pollute the global scope and only use the polyfill in a non-pollutting way:

```
var path = require('../../internals/path');

var Object = path.Object;
```

This is why you get an error that `path.Object` is undefined, the code that modified `path` is not loaded. Now, why that code is not loaded? Because if you look in the webpack bundle you find this as the ` "./node_modules/core-js-pure/modules/es.object.define-property.js"` module:

```
 /***/ "./node_modules/core-js-pure/modules/es.object.define-property.js":
      /*!************************************************************************!*\
  !*** ./node_modules/core-js-pure/modules/es.object.define-property.js ***!
  \************************************************************************/
      /*! exports provided: default */
      /***/ function (module, __webpack_exports__, __webpack_require__) {
        "use strict";
        __webpack_require__.r(__webpack_exports__);
        /* harmony default export */ __webpack_exports__["default"] =
          __webpack_require__.p + "font/1af590cde799e24f8390fc51e79dcbf1.js";

        /***/
      }
```

Webpack and `file-loader` put it in the font folder. And finally, that is because you have this test in your webpack config:

```
{
    test:/\.[svg|ttf|woff|eot]/i,
    use:[{
        loader:'file-loader',
        options:{
            outputPath:'font/'
        }
    }]
}
```

Change `test:/\.[svg|ttf|woff|eot]/i` to `test: /\.(svg|ttf|woff|eot)$/i` and everything will work as expected. You will notice that the `font` folder doesn't have `.js` files anymore, it only includes the fonts as it should.

Thanks for reading :)
